### ISSUE TITLE
Bazel: Update dependencies hosted on GitLab

### ISSUE THREAD

**Issue by cpsauer** [renovatebot/renovate#7945]
**What would you like Renovate to be able to do?**

It'd be awesome if Renovate could update GitLab-hosted Bazel dependencies in the same way it does GitHub-hosted ones.
  

**Did you already have any implementation ideas?**

Currently Renovate seems to be GitHub-only, with a filter happening in the lines below. In the Renovate log, this manifests as  GitLab dependencies failing with  "Failed to find dependency in bazel WORKSPACE definition".
https://github.com/renovatebot/renovate/blob/c63ffa5d6232da8e68186389561531ac8f3c5943/lib/manager/bazel/extract.ts#L24-L26
I assume we'd also need GitLab release, tag, commit fetching logic, parallel to the GitHub case elsewhere, unless it's been implemented for something else. I haven't looked at that part of the codebase yet. Update: `gitlab-tags` datasource already built, says @viceice.

**Other notes**

#7851 and #7852 otherwise together made Renovate work great on all the rest of our (GitHub) dependencies. Only the GitLab minority remain. So, while this is lower priority than those were, adding GitLab support would take us to full dependency update automation with Renovate (!).

Here's an example of a GitLab-hosted Bazel dependency, in case it's helpful. It's for a popular C++ linear algebra library.
```
http_archive(
    name = "eigen",
    build_file = "@//Bazel:eigen.BUILD",
    url = "https://gitlab.com/libeigen/eigen/-/archive/3.3.8/eigen-3.3.8.tar.gz",
    sha256 = "146a480b8ed1fb6ac7cd33fec9eb5e8f8f62c3683b3f850094d9d5c35a92419a",
    strip_prefix = "eigen-3.3.8",
)
```

Related issue: #6847 starts with a general request for http_archives from other, non github sources but eventually gets scoped toward Go.

Thanks for considering!
Chris

**Comment by viceice** [comment]
We already have a `gitlab-tags` datasource.

**Comment by cpsauer** [comment]
Oh, sweet! Thanks, @viceice, for this, and the speedy review last night :)

**Comment by rarkins** [comment]
Blocked by #9667

**Comment by cpsauer** [comment]
Sweet! Again, thank you all so much for all these wonderful Bazel improvements, @zharinov, @viceice, @rarkins! I know I might be the most vocal, but I'm speaking on behalf of my team + the users of our Bazel open source projects. Really appreciate it! Thank you! 🚀
